packages/smart-contracts/scripts-create2/contract-setup/update-fee-proxies.ts

```diff
@@ -0,0 +1,85 @@
+import {
+  singleRequestProxyFactoryArtifact,
+  erc20FeeProxyArtifact,
+  ethereumFeeProxyArtifact,
+} from '../../src/lib';
+import { HardhatRuntimeEnvironmentExtended } from '../types';
+import { getSignerAndGasFees } from './adminTasks';
+import { EvmChains } from '@requestnetwork/currency';
+import { executeContractMethod } from './execute-contract-method';
+import { Contract } from 'ethers';
+
+/**
+ * Update the proxy addresses in the SingleRequestProxyFactory contract
+ * @param hre Hardhat runtime environment
+ * @param signWithEoa Are transactions to be signed by an EOA
+ */
+export const updateFeeProxies = async (
+  hre: HardhatRuntimeEnvironmentExtended,
+  signWithEoa: boolean,
+): Promise<void> => {
+  for (const network of hre.config.xdeploy.networks) {
+    try {
+      EvmChains.assertChainSupported(network);
+
+      const factoryAddress = singleRequestProxyFactoryArtifact.getAddress(network);
+      const erc20ProxyAddress = erc20FeeProxyArtifact.getAddress(network);
+      const ethereumProxyAddress = ethereumFeeProxyArtifact.getAddress(network);
+
+      if (!factoryAddress || !erc20ProxyAddress || !ethereumProxyAddress) {
+        console.info(`Missing contract deployment on ${network}`);
+        continue;
+      }
+
+      const { signer, txOverrides } = await getSignerAndGasFees(network, hre);
+
+      const factory = new Contract(factoryAddress, factoryAbi, signer);
+
+      // Check current values
+      const currentErc20Proxy = await factory.erc20FeeProxy();
+      const currentEthereumProxy = await factory.ethereumFeeProxy();
+
+      // Update ERC20 proxy if needed
+      if (currentErc20Proxy.toLowerCase() !== erc20ProxyAddress.toLowerCase()) {
+        await executeContractMethod({
+          network,
+          contract: factory,
+          method: 'setERC20FeeProxy',
+          props: [erc20ProxyAddress],
+          txOverrides,
+          signer,
+          signWithEoa,
+        });
+        console.log(`Updated ERC20FeeProxy to ${erc20ProxyAddress} on ${network}`);
+      } else {
+        console.log(`ERC20FeeProxy is already set to ${erc20ProxyAddress} on ${network}`);
+      }
+
+      // Update Ethereum proxy if needed
+      if (currentEthereumProxy.toLowerCase() !== ethereumProxyAddress.toLowerCase()) {
+        await executeContractMethod({
+          network,
+          contract: factory,
+          method: 'setEthereumFeeProxy',
+          props: [ethereumProxyAddress],
+          txOverrides,
+          signer,
+          signWithEoa,
+        });
```

Comment:
Can you move this logic to a function in the [adminTasks.ts](https://github.com/RequestNetwork/requestNetwork/blob/5c83706a63dd6c6b6a34bcd6d2d7752fb825f596/packages/smart-contracts/scripts-create2/contract-setup/adminTasks.ts) file?